tf-psa-crypto/core/psa_crypto.c

```diff
@@ -7965,7 +7988,13 @@ psa_status_t psa_generate_key_ext(const psa_key_attributes_t *attributes,
      * storage ( thus not in the case of generating a key in a secure element
      * with storage ( MBEDTLS_PSA_CRYPTO_SE_C ) ),we have to allocate a
      * buffer to hold the generated key material. */
-    if (slot->key.data == NULL) {
+#if defined(MBEDTLS_PSA_STATIC_KEY_SLOTS)
+    int is_slot_unused = (slot->key.in_use == 0);
+#else
+    int is_slot_unused = (slot->key.data == NULL);
+#endif
+
+    if (is_slot_unused) {
```

Comment:
As in `psa_allocate_buffer_to_slot` above, I'm pretty sure this test is always true and I've checked that the else path is not triggered by our test suite in the default configuration.